berserker-core/src/main/java/io/smartcat/berserker/worker/AsyncWorker.java

```diff
@@ -20,107 +23,96 @@
-        this(delegate, queueCapacity, workerStatsGetherer, dropFromHead, threadCount, new DefaultThreadFactory());
+    public AsyncWorker(Worker<T> delegate, int queueCapacity, boolean dropFromHead, int threadCount) {
+        this(delegate, queueCapacity, dropFromHead, threadCount, new DefaultThreadFactory());
     }
 
     /**
      * Constructs asynchronous worker with specified <code>delegate</code> worker, <code>queueCapacity</code>,
-     * <code>workerStatsGetherer</code>, <code>dropFromHead</code>, <code>threadCount</code> and
-     * <code>threadFactory</code>.
+     * <code>dropFromHead</code>, <code>threadCount</code> and <code>threadFactory</code>.
      *
      * @param delegate Worker which is run in thread pool and to which work is delegated.
      * @param queueCapacity Capacity of the queue used as a packet buffer, must be positive number.
-     * @param workerStatsGetherer Consumer which receives {@link WorkerMeta} packet information.
      * @param dropFromHead If true, packet from head of the queue will be dropped, if false, packet from tail of the
      *            queue will be dropped.
      * @param threadCount Number of thread to be used by thread pool, must be positive number.
      * @param threadFactory ThreadFactory to be used in creating threads for thread pool.
      */
-    public AsyncWorker(Worker<T> delegate, int queueCapacity, Consumer<WorkerMeta<T>> workerStatsGetherer,
-            boolean dropFromHead, int threadCount, ThreadFactory threadFactory) {
+    public AsyncWorker(Worker<T> delegate, int queueCapacity, boolean dropFromHead, int threadCount,
+            ThreadFactory threadFactory) {
         if (delegate == null) {
             throw new IllegalArgumentException("Delegate cannot be null.");
         }
         if (queueCapacity <= 0) {
             throw new IllegalArgumentException("Queue capacity must be positive number.");
         }
-        if (workerStatsGetherer == null) {
-            throw new IllegalArgumentException("Worker stats gatherer cannot be null.");
-        }
         if (threadCount <= 0) {
             throw new IllegalArgumentException("Thread count must be positive number.");
         }
         if (threadFactory == null) {
             throw new IllegalArgumentException("Thread factory cannot be null.");
         }
-        this.workerStatsGetherer = workerStatsGetherer;
         this.queue = new LinkedEvictingBlockingQueue<>(dropFromHead, queueCapacity);
-        this.threadPoolExecutor = createAndInitThreadPoolExecutor(delegate, workerStatsGetherer, threadCount,
-                threadFactory);
+        this.threadPoolExecutor = createAndInitThreadPoolExecutor(delegate, threadCount, threadFactory);
+        this.metricRegistry = new MetricRegistry();
+        this.droppedMeter = metricRegistry.meter("dropped");
+        this.waitTime = metricRegistry.histogram("wait.time");
+        this.serviceTime = metricRegistry.histogram("service.time");
```

Comment:
Names as `service.time` are too generic - once such metrics ends in a metrics collector (InfluxDB for example) it is hard to distinguish the data source. There needs to be a mechanism (e.g. through configuration) to specify a metrics name prefix. Also, a default value for that prefix should exists - for example, `io.smartcat.berserker`.